tests/create_task_ignore_error.yml

```diff
@@ -0,0 +1,39 @@
+name: Create Tes Task Job With Ignore Error Flag
+description: Job to create a new TES Task with ignore_error flag
+service: TES
+compatible_versions:
+  - v1.1
+tags:
+  - logical
```

Comment:
A functional variant of this test could be quite easily provided as well, no?

Reply:
Regarding the functionality, we would need to check if the individual executors were successful or not. From the executor logs, this might not be possible and `tesTask.state` might be the only way. I earlier thought that if `ignore_error` was `true`, the state would be `COMPLETE`. However, now I looked at the specs again. There is no such mention. Enabling the flag would record the exit codes and move on to the next executor. For the task state, it should be logged as `EXECUTOR_ERROR`, according to the specs, if even a single one of them failed.

> `EXECUTOR_ERROR`: The task encountered an error in one of the Executor processes. Generally, this means that an Executor exited with a non-zero exit code.

Lets look at our scenarios.
1. Successful Task (Positive) - It will pass for both values of `ignore_error` flag and state will be `COMPLETE`
2. Failed Task (Negative) - If the state is `EXECUTOR_ERROR` (due to above logic) for both values of `ignore_error` flag, `ignore_error` again plays no role.

Therefore, this test just acts as a schema check on our side.

Reply:
Also, @uniqueg please have a look at my comments in #29 if there is something you find relevant.

Reply:
Very good comments in the closed PR, I've added comments there as well.

As for this particular issue: It's true that `EXECUTOR_ERROR` is expected in either case. (The reason for adding the flag in 1.1.0 is to support use cases like a tear down step, which should always be executed, even if the main tasks fail)

The difference is that in the default behavior (`ignore_error: false`), the second executor is never even started. I think we should be able to tell that by the number of executor logs we get.

Reply:
Right, this is one case through which we test this feature. However, there is no functionality in the suite yet to check the _array size_ of a particular response field. It can be added later. Though it seems related only to TES right now, it might be of some use to later specs. We would add stricter tests but it would make the test files too complicated.

Reply:
Seems a reasonable feature to support, in my opinion. I mean, so far we are using the runner for a total of one specification, and we already have a use case for that feature. That's 1 out of 1, right? :laughing: 

But sure, certainly a separate issue if not yet supported.

Reply:
Sure, it will be a good feature to add.

Reply:
This feature will be added in #33 . Perhaps we can leave this test out for now and add later.